torch/csrc/lazy/core/tensor.cpp

```diff
@@ -345,7 +347,9 @@ std::shared_ptr<LazyView> LazyTensor::CreateView(ViewInfo view_info) const {
 }
 
 LazyTensorPtr LazyTensor::CreateViewTensor(ViewInfo view_info) const {
-  return Create(CreateView(std::move(view_info)), GetDevice());
+  auto new_tensor = Create(CreateView(std::move(view_info)), GetDevice());
+  new_tensor->storage_ = Storage();
```

Comment:
hmm, so we explicitly copy the Storage from one LT to another one if we know we are creating a view.

i was actually thinking we could tie it into the 'data()' object in some way, so if 2 LazyTensors happen to point to the same IR value then they also share the same storage.  I'm not actually sure if that is a desirable property, and maybe your approach is good enough.  It certainly seems to be simpler, which is nice.

Reply:
> if 2 LazyTensors happen to point to the same IR value then they also share the same storage

just curious - are you saying it's possible to be in a state where you have two LTC tensor's that aren't views of each other, but they actually point to the same IR under the hood?